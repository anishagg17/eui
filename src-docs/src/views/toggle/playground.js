import { PropTypes } from 'react-view';
import { EuiToggle } from '../../../../src/components/';
import propUtilityForPlayground from '../../services/playground/props';

export default () => {
  const docgenInfo = Array.isArray(EuiToggle.__docgenInfo)
    ? EuiToggle.__docgenInfo[0]
    : EuiToggle.__docgenInfo;
  const propsToUse = propUtilityForPlayground(docgenInfo.props);

  propsToUse.label.value = 'Is toggle on?';
  propsToUse.checked.stateful = true;
  propsToUse.children = {
    ...propsToUse.children,
    type: PropTypes.ReactNode,
    // eslint-disable-next-line quotes
    value: `{checked ? 'On' : 'Off'}`,
    hidden: true,
  };
  propsToUse.onChange = {
    ...propsToUse.onChange,
    type: PropTypes.Function,
    // eslint-disable-next-line quotes
    value: 'e => setChecked(e.target.checked)',
    propHook: {
      what: 'e.target.checked',
      into: 'checked',
    },
  };

  return {
    componentName: 'EuiToggle',
    props: propsToUse,
    scope: {
      EuiToggle,
    },
    imports: {
      '@elastic/eui': {
        named: ['EuiToggle'],
      },
    },
  };
};
