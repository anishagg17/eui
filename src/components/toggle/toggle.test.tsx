import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { EuiToggle } from './toggle';

describe('EuiToggle', () => {
  test('is rendered', () => {
    const component = render(
      <EuiToggle label="Is toggle on?" {...requiredProps} />
    );

    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    test('is disabled', () => {
      const component = render(
        <EuiToggle label="Is toggle on?" isDisabled {...requiredProps} />
      );

      expect(component).toMatchSnapshot();
    });

    test('is rendered with onChange provided', () => {
      const component = render(
        <EuiToggle
          label="Is toggle on?"
          onChange={jest.fn()}
          {...requiredProps}
        />
      );

      expect(component).toMatchSnapshot();
    });

    test('is checked', () => {
      const component = render(
        <EuiToggle
          label="Is toggle on?"
          checked
          onChange={jest.fn()}
          {...requiredProps}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
