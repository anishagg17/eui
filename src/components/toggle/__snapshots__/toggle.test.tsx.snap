// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiToggle is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiToggle testClass1 testClass2"
>
  <input
    aria-label="Is toggle on?"
    class="euiToggle__input"
    data-test-subj="test subject string"
    type="checkbox"
  />
</div>
`;

exports[`EuiToggle props is checked 1`] = `
<div
  aria-label="aria-label"
  class="euiToggle euiToggle--checked testClass1 testClass2"
>
  <input
    aria-label="Is toggle on?"
    checked=""
    class="euiToggle__input"
    data-test-subj="test subject string"
    type="checkbox"
  />
</div>
`;

exports[`EuiToggle props is disabled 1`] = `
<div
  aria-label="aria-label"
  class="euiToggle testClass1 testClass2"
>
  <input
    aria-label="Is toggle on?"
    class="euiToggle__input"
    data-test-subj="test subject string"
    disabled=""
    type="checkbox"
  />
</div>
`;

exports[`EuiToggle props is rendered with onChange provided 1`] = `
<div
  aria-label="aria-label"
  class="euiToggle testClass1 testClass2"
>
  <input
    aria-label="Is toggle on?"
    class="euiToggle__input"
    data-test-subj="test subject string"
    type="checkbox"
  />
</div>
`;
